cpp/src/reshape/byte_cast.cu

```diff
@@ -0,0 +1,209 @@
+   * mimics Spark's cast to binary type.
+   */
+  struct flip_endianness_lambda {
+    char* d_chars;
+    const char* d_data;
+    uint32_t mask;
+    __device__ void operator()(int byte_index)
+    {
+      d_chars[byte_index] = d_data[byte_index + mask - ((byte_index & mask) << 1)];
+    }
+  };
+
+  template <typename T>
+  struct normalize_lambda {
+    T* d_result;
+    const T* d_data;
+    __device__ void operator()(int index)
+    {
+      auto e = d_data[index];
+      if (isnan(e)) {
+        d_result[index] = std::numeric_limits<T>::quiet_NaN();
+      } else if (T{0.0} == e) {
+        d_result[index] = T{0.0};
+      } else
+        d_result[index] = e;
+    }
+  };
+
+  template <typename T>
+  struct normalize_flip_endianess_lambda {
+    char* d_chars;
+    const T* d_data;
+    size_t size;
+    __device__ void operator()(int index)
+    {
+      T normal           = normalize(d_data[index]);
+      char* normal_bytes = reinterpret_cast<char*>(&normal);
+      for (int i = 0; i < size; i++) { d_chars[index * size + i] = normal_bytes[size - 1 - i]; }
+    }
+
+    __device__ T normalize(T in)
+    {
+      if (isnan(in)) { return std::numeric_limits<T>::quiet_NaN(); }
+      if (T{0.0} == in) { return T{0.0}; }
+      return in;
+    }
+  };
+
+  template <typename T,
+            std::enable_if_t<!std::is_integral<T>::value and !is_floating_point<T>()>* = nullptr>
+  std::unique_ptr<column> operator()(column_view const& input_column,
+                                     flip_endianness configuration,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream) const
+  {
+    CUDF_FAIL("Unsupported non-numeric and non-string column");
+  }
+
+  template <typename T, std::enable_if_t<is_floating_point<T>()>* = nullptr>
+  std::unique_ptr<column> operator()(column_view const& input_column,
+                                     flip_endianness configuration,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream) const
+  {
+    auto byte_column = make_numeric_column(data_type{type_id::UINT8},
+                                           input_column.size() * cudf::size_of(input_column.type()),
+                                           mask_state::UNALLOCATED,
+                                           stream,
+                                           mr);
+
+    if (configuration == flip_endianness::YES) {
+      thrust::for_each(rmm::exec_policy(stream)->on(stream),
+                       thrust::make_counting_iterator(0),
+                       thrust::make_counting_iterator(input_column.size()),
+                       normalize_flip_endianess_lambda<T>{
+                         reinterpret_cast<char*>(byte_column->mutable_view().data<T>()),
+                         input_column.data<T>(),
+                         cudf::size_of(input_column.type())});
```

Comment:
```suggestion
                         sizeof(T)});
```
The `cudf::size_of` will do another `type_dispatcher` call which we do not need since we are currently in a `type_dispatcher` call which has already resolved `T` from the `input_column` type.